addons/knowledge/static/src/components/emoji_picker/emoji_picker.js

```diff
@@ -0,0 +1,62 @@
+/** @odoo-module **/
+
+import emojis from '@mail/js/emojis';
+const { Component, useSubEnv } = owl;
+
+class Emoji extends Component {
+    /**
+     * @override
+     */
+    setup () {
+        super.setup();
+        // Mock the template variables:
+        this.className = '';
+        this.emojiListView = {
+            /**
+             * @param {Event} event
+             */
+            onClickEmoji: event => {
+                const $target = $(event.target);
+                this.env.onClickEmoji($target.data('unicode'));
+            },
+        };
+    }
+}
+
+Emoji.template = 'mail.Emoji';
+Emoji.props =  {
+    emoji: Object,
+    emojiListViewLocalId: String,
+};
+
+class EmojiPicker extends Component {
+    /**
+     * @override
+     */
+    setup () {
+        super.setup();
+        // Mock the template variables:
+        this.emojis = emojis;
+        this.className = '';
+        this.emojiListView = {
+            localId: '',
+        };
+        // Set up sub-environment variables:
+        useSubEnv({
+            onClickEmoji: this.props.onClickEmoji
```

Comment:
really weird to use a subenv. why dont you just propagate the onclickemoji handler as a prop?